pipert/contrib/routines/message_to_redis.py

```diff
@@ -25,7 +25,11 @@ def main_logic(self, *args, **kwargs):
         try:
             msg = self.message_queue.get(block=False)
             msg.record_exit(self.component_name, self.logger)
-            encoded_msg = message_encode(msg)
+            if self.use_memory and isinstance(msg.payload, FramePayload):
+                encoded_msg = message_encode(msg,
+                                             generator=self.generator)
+            else:
+                encoded_msg = message_encode(msg)
```

Comment:
Why using the shared memory only for this type of payload ?
And why not sending the generator to the `message_encode` function, I am pretty sure that the function is handling a situation that the generator is None ?